pysilcam/tracking/track.py

```diff
@@ -0,0 +1,417 @@
+from docopt import docopt
+from pysilcam.tracking.silcamtracker import *
+import numpy as np
+import sys
+import glob
+import pandas as pd
+from scipy import interpolate
+import matplotlib.pyplot as plt
+from skimage.transform import rotate
+from pysilcam.postprocess import explode_contrast
+from pysilcam.config import PySilcamSettings, settings_from_h5
+
+
+def make_output_path(datapath):
+    dataset_name = os.path.split(datapath)[1]
+    outputpath = os.path.join(datapath, ('../output_' + dataset_name))
+    os.makedirs(outputpath, exist_ok=True)
+    return outputpath
+
+
+def plot_boxplot(dataset_names, tracks, PIX_SIZE, figurename):
+    ps = np.arange(0, len(dataset_names))
+    ls = dataset_names
+
+    f, a = plt.subplots(3, 1, figsize=(6, 12))
+
+    plt.sca(a[0])
+    box_data = [tracks[dataset_names[i]]['width'] /
+                tracks[dataset_names[i]]['length']
+                for i in range(len(dataset_names))]
+    plt.boxplot(box_data, positions=ps, labels=ls)
+    plt.ylabel('Minor/Major axis')
+    plt.ylim(0, 1)
+    plt.gca().xaxis.tick_top()
+    plt.xticks(rotation=45, horizontalalignment='left')
+
+    plt.sca(a[1])
+    box_data = [tracks[dataset_names[i]]['length'] * PIX_SIZE / 1000
+                for i in range(len(dataset_names))]
+    plt.boxplot(box_data, positions=ps, labels=ls)
+    plt.ylabel('Maxjor Axis Length [mm]')
+    plt.xticks([])
+    plt.ylim(0, 12)
+
+    plt.sca(a[2])
+    box_data = [tracks[dataset_names[i]]['S_cms'] * 10
+                for i in range(len(dataset_names))]
+
+    plt.boxplot(box_data, positions=ps, labels=ls)
+    plt.ylabel('Net speed [mm/s]')
+    plt.xticks(rotation=45, horizontalalignment='right')
+    plt.ylim(0, 2.5)
+
+    figurename = os.path.join(figurename + '.png')
+    print('  saving:', figurename)
+    plt.savefig(figurename, dpi=600, bbox_inches='tight')
+    print('  saved')
+
+
+def im_from_timestamp(timestamp, rawdatapath):
+    searchfilename = timestamp.strftime('D%Y%m%dT%H%M%S.%f')
+    hits = glob.glob(os.path.join(rawdatapath, searchfilename + '*'))
+    imagename = hits[0]
+    im = silcam_load(imagename)
```

Comment:
F405: 'silcam_load' may be undefined, or defined from star imports: pysilcam.tracking.silcamtracker  <!-- Automatically posted by Lintly -->